### PR TITLE
CLN: remove unnecessary trailing commas on issues #35925

### PR DESCRIPTION
**Pull request by satrio-hw**
#35925

- pandas/tests/arrays/categorical/test_replace.py
- pandas/tests/arrays/sparse/test_array.py
- pandas/tests/arrays/test_array.py
- pandas/tests/arrays/test_timedeltas.py
- pandas/tests/base/test_conversion.py
- pandas/tests/dtypes/test_missing.py
- pandas/tests/extension/base/methods.py
- pandas/tests/extension/test_sparse.py
- pandas/tests/frame/indexing/test_setitem.py

please give me a feedback if there is something that I have to change.. thank you before

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Hi @satrio-hw 

You will need to run `black` again on the files you've changed to make sure they won't be changed

**Comment by satrio-hw** [comment]
@MarcoGorelli I've tried to rerun black on modified files, some of them revert back with commas on it.. And still got some error when chacking after I push back.. Something I could do to fix this?

File: pandas/tests/arrays/sparse/test_array.py [hunk 1]
```diff
@@ -696,14 +694,23 @@ def test_getslice_tuple(self):
         res = sparse[
             4:,
         ]  # noqa: E231
-        exp = SparseArray(dense[4:,])  # noqa: E231
+        exp = SparseArray(
+            dense[
+                4:,
+            ]
+        )  # noqa: E231
```
Comment (MarcoGorelli) [inline pandas/tests/arrays/sparse/test_array.py:701]: Seems the two versions of `black` disagree on how to format this...I'd say you can remove this file from this PR and we'll take a look at it later, the rest of your changes look good

Comment (MarcoGorelli) [reply pandas/tests/arrays/sparse/test_array.py:701]: @satrio-hw actually, it should be fine to remove this comma (as with it the `noqa` comment):

```
>>> sparse[4:,]
[0.0, 5.0, nan, nan, 0.0]
Fill: nan
IntIndex
Indices: array([0, 1, 4], dtype=int32)

>>> sparse[4:]
[0.0, 5.0, nan, nan, 0.0]
Fill: nan
IntIndex
Indices: array([0, 1, 4], dtype=int32)
```

**Comment by MarcoGorelli** [comment]
@satrio-hw I think you can not include pandas/tests/arrays/sparse/test_array.py as part of this pull request, as the two versions of `black` will not agree on how to reformat it and we need to the trailing comma there. So, we'll update it when we update the version of black. The rest of your PR looks good!

**Comment by satrio-hw** [comment]
@MarcoGorelli I've revert back pandas/tests/arrays/sparse/test_array.py to origin/master. To make sure I've run a command `./ci/code_checks.sh lint` and check there is only one changes to commit.. linting check done successful after pushing changes, but now I got failed tests on pandas-dev.pandas

is there something wrong in the procedures that I've done? thanks before

**Review by MarcoGorelli — APPROVED** [review]
This looks correct now, thanks, the CI error should be unrelated - can you fetch and merge master, then push, to restart the CI process and get it green?

**Comment by satrio-hw** [comment]
@MarcoGorelli I've fetch and merge master, and push it again.. now all green.. thank you very much for your guidance

**Comment by jreback** [comment]
thanks @satrio-hw

**Pull request merged**